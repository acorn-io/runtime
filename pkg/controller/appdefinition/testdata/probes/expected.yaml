kind: Deployment
apiVersion: apps/v1
metadata:
  name: oneimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "oneimage"
      "acorn.io/managed": "true"
  template:
    metadata:
      labels:
        "acorn.io/app-namespace": "app-namespace"
        "acorn.io/app-name": "app-name"
        "acorn.io/app-public-name": "app-name"
        "acorn.io/container-name": "oneimage"
        "acorn.io/managed": "true"
      annotations:
        acorn.io/container-spec: '{"image":"image-name","ports":[{"port":80,"protocol":"http","targetPort":81}],"probes":null,"sidecars":{"left":{"image":"foo","probes":[{"http":{"headers":{"foo":"bar"},"url":"http://localhost/foo/bar"},"type":"readiness"},{"tcp":{"url":"garbage://1.1.1.1:1234/foo/bar"},"type":"startup"},{"exec":{"command":["/bin/true"]},"type":"liveness"}]}}}'
    spec:
      terminationGracePeriodSeconds: 5
      enableServiceLinks: false
      serviceAccountName: oneimage
      hostname: oneimage
      imagePullSecrets:
        - name: oneimage-pull-1234567890ab
      containers:
        - name: oneimage
          image: "image-name"
          ports:
          - containerPort: 81
            protocol: "TCP"
          readinessProbe:
            tcpSocket:
              port: 81
        - name: left
          image: "foo"
          readinessProbe:
            httpGet:
              port: 80
              path: "/foo/bar"
              httpHeaders:
                - name: foo
                  value: bar
          livenessProbe:
            exec:
              command:
                - /bin/true
          startupProbe:
            tcpSocket:
              port: 1234
              host: 1.1.1.1
---
kind: PodDisruptionBudget
apiVersion: policy/v1
metadata:
  name: oneimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "oneimage"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "oneimage"
      "acorn.io/managed": "true"
  maxUnavailable: 25%
---
kind: ServiceInstance
apiVersion: internal.acorn.io/v1
metadata:
  name: oneimage
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    "acorn.io/container-name": "oneimage"
    "acorn.io/public-name": "app-name.oneimage"
spec:
  appName: app-name
  appNamespace: app-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    "acorn.io/container-name": "oneimage"
  type: ClusterIP
  container: oneimage
  ports:
    - port: 80
      targetPort: 81
      protocol: http
      appProtocol: "HTTP"
      name: "80"

---
kind: Secret
apiVersion: v1
metadata:
  name: oneimage-pull-1234567890ab
  namespace: app-created-namespace
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
type: "kubernetes.io/dockerconfigjson"
data:
  ".dockerconfigjson": eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
---
kind: Secret
apiVersion: v1
metadata:
  name: nodefault-pull-1234567890ab
  namespace: app-created-namespace
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
type: "kubernetes.io/dockerconfigjson"
data:
  ".dockerconfigjson": eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
---
kind: AppInstance
apiVersion: internal.acorn.io/v1
metadata:
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
status:
  namespace: app-created-namespace
  appImage:
    id: test
  appSpec:
    containers:
      nodefault:
        probes: []
        ports:
          - port: 80
            targetPort: 81
            protocol: http
        image: "image-name"
      oneimage:
        sidecars:
          left:
            image: "foo"
            probes:
              - type: "readiness"
                http:
                  url: "http://localhost/foo/bar"
                  headers:
                    foo: bar
              - type: "startup"
                tcp:
                  url: "garbage://1.1.1.1:1234/foo/bar"
              - type: "liveness"
                exec:
                  command: ["/bin/true"]
        ports:
          - port: 80
            targetPort: 81
            protocol: http
        image: "image-name"
  conditions:
    - type: defined
      reason: Success
      status: "True"
      success: true

---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: nodefault
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "nodefault"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "nodefault"
      "acorn.io/managed": "true"
  template:
    metadata:
      labels:
        "acorn.io/app-namespace": "app-namespace"
        "acorn.io/app-name": "app-name"
        "acorn.io/app-public-name": "app-name"
        "acorn.io/container-name": "nodefault"
        "acorn.io/managed": "true"
      annotations:
        acorn.io/container-spec: '{"image":"image-name","ports":[{"port":80,"protocol":"http","targetPort":81}],"probes":[]}'
    spec:
      terminationGracePeriodSeconds: 5
      enableServiceLinks: false
      serviceAccountName: nodefault
      hostname: nodefault
      imagePullSecrets:
        - name: nodefault-pull-1234567890ab
      containers:
        - name: nodefault
          image: "image-name"
          ports:
            - containerPort: 81
              protocol: "TCP"
---
kind: PodDisruptionBudget
apiVersion: policy/v1
metadata:
  name: nodefault
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/container-name": "nodefault"
    "acorn.io/managed": "true"
spec:
  selector:
    matchLabels:
      "acorn.io/app-namespace": "app-namespace"
      "acorn.io/app-name": "app-name"
      "acorn.io/container-name": "nodefault"
      "acorn.io/managed": "true"
  maxUnavailable: 25%
---
kind: ServiceInstance
apiVersion: internal.acorn.io/v1
metadata:
  name: nodefault
  namespace: app-created-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    "acorn.io/container-name": "nodefault"
    "acorn.io/public-name": "app-name.nodefault"
spec:
  appName: app-name
  appNamespace: app-namespace
  labels:
    "acorn.io/app-namespace": "app-namespace"
    "acorn.io/app-name": "app-name"
    "acorn.io/managed": "true"
    "acorn.io/container-name": "nodefault"
  container: nodefault
  ports:
    - port: 80
      targetPort: 81
      protocol: http
      appProtocol: "HTTP"
      name: "80"