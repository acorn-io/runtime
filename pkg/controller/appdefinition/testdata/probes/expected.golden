`apiVersion: v1
kind: ServiceAccount
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: nodefault
    acorn.io/managed: "true"
  name: nodefault
  namespace: app-created-namespace

---
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: nodefault
    acorn.io/managed: "true"
  name: nodefault
  namespace: app-created-namespace
spec:
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: nodefault
      acorn.io/managed: "true"
  strategy: {}
  template:
    metadata:
      annotations:
        acorn.io/container-spec: '{"image":"image-name","metrics":{},"ports":[{"port":80,"protocol":"http","targetPort":81}],"probes":[]}'
      creationTimestamp: null
      labels:
        acorn.io/app-name: app-name
        acorn.io/app-namespace: app-namespace
        acorn.io/app-public-name: app-name
        acorn.io/container-name: nodefault
        acorn.io/managed: "true"
    spec:
      containers:
      - image: image-name
        name: nodefault
        ports:
        - containerPort: 81
          protocol: TCP
        resources: {}
      enableServiceLinks: false
      hostname: nodefault
      imagePullSecrets:
      - name: nodefault-pull-1234567890ab
      serviceAccountName: nodefault
      terminationGracePeriodSeconds: 5
status: {}

---
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: nodefault
    acorn.io/managed: "true"
  name: nodefault
  namespace: app-created-namespace
spec:
  maxUnavailable: 25%
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: nodefault
      acorn.io/managed: "true"
status:
  currentHealthy: 0
  desiredHealthy: 0
  disruptionsAllowed: 0
  expectedPods: 0

---
apiVersion: v1
kind: ServiceAccount
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
  name: oneimage
  namespace: app-created-namespace

---
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
  name: oneimage
  namespace: app-created-namespace
spec:
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: oneimage
      acorn.io/managed: "true"
  strategy: {}
  template:
    metadata:
      annotations:
        acorn.io/container-spec: '{"image":"image-name","metrics":{},"ports":[{"port":80,"protocol":"http","targetPort":81}],"probes":null,"sidecars":{"left":{"image":"foo","metrics":{},"probes":[{"http":{"headers":{"foo":"bar"},"url":"http://localhost/foo/bar"},"type":"readiness"},{"tcp":{"url":"garbage://1.1.1.1:1234/foo/bar"},"type":"startup"},{"exec":{"command":["/bin/true"]},"type":"liveness"}]}}}'
      creationTimestamp: null
      labels:
        acorn.io/app-name: app-name
        acorn.io/app-namespace: app-namespace
        acorn.io/app-public-name: app-name
        acorn.io/container-name: oneimage
        acorn.io/managed: "true"
    spec:
      containers:
      - image: image-name
        name: oneimage
        ports:
        - containerPort: 81
          protocol: TCP
        readinessProbe:
          tcpSocket:
            port: 81
        resources: {}
      - image: foo
        livenessProbe:
          exec:
            command:
            - /bin/true
        name: left
        readinessProbe:
          httpGet:
            httpHeaders:
            - name: foo
              value: bar
            path: /foo/bar
            port: 80
        resources: {}
        startupProbe:
          tcpSocket:
            host: 1.1.1.1
            port: 1234
      enableServiceLinks: false
      hostname: oneimage
      imagePullSecrets:
      - name: oneimage-pull-1234567890ab
      serviceAccountName: oneimage
      terminationGracePeriodSeconds: 5
status: {}

---
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
  name: oneimage
  namespace: app-created-namespace
spec:
  maxUnavailable: 25%
  selector:
    matchLabels:
      acorn.io/app-name: app-name
      acorn.io/app-namespace: app-namespace
      acorn.io/container-name: oneimage
      acorn.io/managed: "true"
status:
  currentHealthy: 0
  desiredHealthy: 0
  disruptionsAllowed: 0
  expectedPods: 0

---
apiVersion: internal.acorn.io/v1
kind: ServiceInstance
metadata:
  annotations:
    acorn.io/app-generation: "0"
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: nodefault
    acorn.io/managed: "true"
    acorn.io/public-name: app-name.nodefault
  name: nodefault
  namespace: app-created-namespace
spec:
  appName: app-name
  appNamespace: app-namespace
  container: nodefault
  default: false
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: nodefault
    acorn.io/managed: "true"
  ports:
  - port: 80
    protocol: http
    targetPort: 81
status: {}

---
apiVersion: internal.acorn.io/v1
kind: ServiceInstance
metadata:
  annotations:
    acorn.io/app-generation: "0"
  creationTimestamp: null
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
    acorn.io/public-name: app-name.oneimage
  name: oneimage
  namespace: app-created-namespace
spec:
  appName: app-name
  appNamespace: app-namespace
  container: oneimage
  default: false
  labels:
    acorn.io/app-name: app-name
    acorn.io/app-namespace: app-namespace
    acorn.io/container-name: oneimage
    acorn.io/managed: "true"
  ports:
  - port: 80
    protocol: http
    targetPort: 81
status: {}

---
apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
  name: nodefault-pull-1234567890ab
  namespace: app-created-namespace
type: kubernetes.io/dockerconfigjson

---
apiVersion: v1
data:
  .dockerconfigjson: eyJhdXRocyI6eyJpbmRleC5kb2NrZXIuaW8iOnsiYXV0aCI6Ik9nPT0ifX19
kind: Secret
metadata:
  creationTimestamp: null
  labels:
    acorn.io/managed: "true"
    acorn.io/pull-secret: "true"
  name: oneimage-pull-1234567890ab
  namespace: app-created-namespace
type: kubernetes.io/dockerconfigjson

---
apiVersion: internal.acorn.io/v1
kind: AppInstance
metadata:
  creationTimestamp: null
  name: app-name
  namespace: app-namespace
  uid: 1234567890abcdef
spec:
  image: test
status:
  appImage:
    id: test
    imageData: {}
    vcs: {}
  appSpec:
    containers:
      nodefault:
        image: image-name
        metrics: {}
        ports:
        - port: 80
          protocol: http
          targetPort: 81
        probes: []
      oneimage:
        image: image-name
        metrics: {}
        ports:
        - port: 80
          protocol: http
          targetPort: 81
        probes: null
        sidecars:
          left:
            image: foo
            metrics: {}
            probes:
            - http:
                headers:
                  foo: bar
                url: http://localhost/foo/bar
              type: readiness
            - tcp:
                url: garbage://1.1.1.1:1234/foo/bar
              type: startup
            - exec:
                command:
                - /bin/true
              type: liveness
  appStatus: {}
  columns: {}
  conditions:
    reason: Success
    status: "True"
    success: true
    type: defined
  defaults: {}
  namespace: app-created-namespace
`
